### ISSUE TITLE
Add a note about using `if()` with `index-repeat()`

### ISSUE THREAD

**Issue by adammichaelwood** [opendatakit/docs#679]
From old gotchas doc:

----

indexed-repeat failure before adding 2nd repeat group
Using XLSForm, I have a repeat and use indexed-repeat to call variables from there into the non-repeat section of the form. It works fine when the index is a static number. When the index is based on another (currently undefined, but defined before the indexed-repeat is reached) variable it crashes when transitioning from the first repetition to the second. Note that this crash is before the prompt using the indexed-repeat function has actually been reached.

My solution was to put an if on the index so it always has a value. That fixed it but I wanted to put a heads up here. I hope that is clear, if not let me know. The bottom line is, when calling from a repeat into the non-repeating part of the form:

This doesn't work - indexed-repeat(${name}, ${repeat}, ${id})
This does - indexed-repeat(${name}, ${repeat}, if(${id} != null, ${id}, 1))
Clarification: This odd behavior occurs because whenever you alter a value, all formulas using the value are re-evaluated. Since the ${repeat} changed (you saved and/or created a new repeat group), the formula containing the indexed-repeat is re-evaluated, but ${id} is not yet initialized.

-----

The examples given in `form-repeats` follow this pattern, but it is not explained.
Probably a good idea to add a note mentioning it.

**Comment by adammichaelwood** [comment]
I cannot currently reproduce this issue. It may have been fixed.

**Comment by lognaturel** [comment]
The issue is if the index (last parameter) is a variable. For example, instead of `indexed-repeat(${party_person_name}, ${party_people}, 2)`, ask the user which person's meal they want to look up -- `indexed-repeat(${party_person_name}, ${party_people}, ${chosen_person})`. ${chosen_person} will be null when the form is loaded and the function call will fail.

**Comment by adammichaelwood** [comment]
Oh, I see.

Can you think of a reasonable (not ridiculous and contrived) example of when this pattern would occur?

**Comment by lognaturel** [comment]
I think having the user look something up based on index is something people do. But maybe you'll prefer something like computing the index of the oldest child and asking a question about him/her.

**Comment by adammichaelwood** [comment]
So my contrived example is very contrived, but... I made the the indexed-repeat look up dynamic.
I am still not able to reproduce this issue.

Here is my current form.

[index-if-gotcha.xlsx](https://github.com/opendatakit/docs/files/1961913/index-if-gotcha.xlsx)

**Comment by lognaturel** [comment]
You're right. The index defaults to 1 as of a few months ago -- https://github.com/opendatakit/javarosa/issues/214. I do think that's a good thing to document. The tip should then be to either make the value that will be used as an index required or to use an if statement like the one in the original gotcha. The default of 1 avoids a crash but can lead to unexpected behavior if the value isn't actually set as expected.
